src/nvim/ui.c

```diff
@@ -150,24 +151,23 @@ void ui_resize(int new_width, int new_height)
   UI_CALL(resize, width, height);
 }
 
-void ui_cursor_on(void)
+void ui_busy_start(void)
 {
-  if (!cursor_enabled) {
-    UI_CALL(cursor_on);
-    cursor_enabled = true;
+  if (!(busy++)) {
+    UI_CALL(busy_start);
   }
+  ui_flush();
 }
 
-void ui_cursor_off(void)
+void ui_busy_stop(void)
 {
-  if (full_screen) {
-    if (cursor_enabled) {
-      UI_CALL(cursor_off);
-    }
-    cursor_enabled = false;
+  if (!(--busy)) {
+    UI_CALL(busy_stop);
   }
+  ui_flush();
```

Comment:
This is because I removed the `ui_flush`` calls in `get_keystroke`/`inchar`. I will add them back when I fix this